### DISCUSSION TITLE
Issue with the function create_em

### DISCUSSION THREAD

**Discussion by fji276828719** [ss3sim/ss3sim#406] [Q&A] [answered]
Hi there,

I am a beginner in Stock Synthesis and ss3sim package. I was confused by an issue I met and sought your help hereby. I tried to set up an OM with 11 fleets, including 10 fishery and one survey, meanwhile the fourth fleet and the survey were with CPUE data. I met with an error below when I used the function create_em with the OM above mentioned.

Error in `.rowNamesDF<-`(x, value = value) : 不允许有重复的'row.names'
In addition: Warning message:
non-unique value when setting 'row.names': ‘4_Fishery’

I tried to change the fleet number in **#_Q_setup for fleets with cpue or survey data** of the om.ctl file from 4 to 1, the function create_em then worked with no error. I read example files of codOM in ss3sim, which only contained one fishery and one survey. So I am not sure about if the fishery with CPUE data can only be set to fleet 1 in an OM with multi-fleets?

I would really appreciate your attention and support. Cheers.

**Comment by kellijohnson-NOAA** [comment]
Thank you for posting a question here and for using {ss3sim}. I am pretty sure that the package can accommodate your desired fleet/survey structure but it just might be an error in `create_em()`. It should not matter what fleet number the survey is. For example, the cod model could have the survey as fleet two instead of fleet one and it would not matter. Is there any way you can send me your input files by posting them here or sending them in an email to kelli.johnson@noaa.gov?

**Reply by fji276828719** [reply]
Hi Kelli, I really appreciate your swift response. I have sent input files via email to you. Please help me to check it out. Thank you so much.

**Comment by kellijohnson-NOAA** [comment, marked as answer]
As I mentioned in my email, I am very sorry for the delay in getting to your question. The errors you were encountering had to do with the fact that your Operating Models were not operational. You can check this by trying to run Stock Synthesis on the files that you sent me. I did this and then looked at echoinput.sso to see how far the model was able to get in reading the files. Then, one by one, I fixed the errors that the extecutable was encountering. Please let me know if you need any more assistance. I am happy to help 😀 !

**Reply by fji276828719** [reply]
You helped me a lot, Kelli. Thank you so much!
